src/main/java/de/uniba/dsg/serverless/provider/google/GoogleLogHandler.java

```diff
@@ -0,0 +1,185 @@
+package de.uniba.dsg.serverless.provider.google;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.gax.paging.Page;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Logging.EntryListOption;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload;
+import com.google.cloud.logging.Payload.StringPayload;
+
+import de.uniba.dsg.serverless.model.PerformanceData;
+import de.uniba.dsg.serverless.model.SeMoDeException;
+import de.uniba.dsg.serverless.model.WritableEvent;
+import de.uniba.dsg.serverless.provider.LogHandler;
+
+/**
+ * Provide your Credentials to use the GoogleLogHandler via the Google Cloud SDK
+ * command <i>gcloud auth application-default login</i>.
+ * 
+ */
+public class GoogleLogHandler implements LogHandler {
+
+	private static final DateTimeFormatter QUERY_DATE_FORMATTER = DateTimeFormatter
+			.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+	private final String functionName;
+	private final LocalDateTime startTime;
+	private final LocalDateTime endTime;
+
+	public GoogleLogHandler(String functionName, LocalDateTime startTime, LocalDateTime endTime) {
+		this.functionName = functionName;
+		this.startTime = startTime;
+		this.endTime = endTime;
+	}
+
+	private List<LogEntry> extractLogEntryFromPage(Page<LogEntry> entries) {
+		List<LogEntry> logEntries = new ArrayList<>();
+		for (LogEntry entry : entries.iterateAll()) {
+			logEntries.add(entry);
+		}
+		return logEntries;
+	}
+
+	private List<LogEntry> executeListLogEntries() throws SeMoDeException {
+
+		List<LogEntry> logEntries = new ArrayList<>();
+
+		// project is inferred from the environment
+		LoggingOptions options = LoggingOptions.getDefaultInstance();
+
+		try (Logging logging = options.getService()) {
+
+			String filter = "resource.type=\"cloud_function\" AND " + "resource.labels.function_name = \""
+					+ this.functionName + "\" AND " + "timestamp > \"" + this.startTime.format(QUERY_DATE_FORMATTER)
+					+ "\" AND " + "timestamp < \"" + this.endTime.format(QUERY_DATE_FORMATTER) + "\"";
+
+			EntryListOption entryFilterOption = EntryListOption.filter(filter);
+
+			Page<LogEntry> entries = logging.listLogEntries(entryFilterOption);
+			logEntries.addAll(this.extractLogEntryFromPage(entries));
+
+			// further pages with log entries
+			while (entries.hasNextPage()) {
```

Comment:
As far as I understand it, loading the next pages is redundant:

The underlying REST-API of the google cloud logging service uses pagination. Several requests need to be send where each requests return a page and the token for the next page (see [https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list](https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list)). 

But the Java SDK that is used here does this pagination and loading of the next pages in the background already.

See the documentation: [http://googleapis.github.io/gax-java/1.25.0/apidocs/com/google/api/gax/paging/Page.html?is-external=true](http://googleapis.github.io/gax-java/1.25.0/apidocs/com/google/api/gax/paging/Page.html?is-external=true)
`java.lang.Iterable<ResourceT> iterateAll()`
_Returns an iterable that traverses all of the elements of the underlying data source. The data is fetched lazily page by page, where each page may contain multiple elements. A new page is fetched whenever the elements of any particular page are exhausted."_

`iterateAll()` is called in line 51.